cluster-autoscaler/core/static_autoscaler.go

```diff
@@ -244,6 +249,11 @@ func (a *StaticAutoscaler) RunOnce(currentTime time.Time) errors.AutoscalerError
 		glog.V(1).Info("No unschedulable pods")
 	} else if a.MaxNodesTotal > 0 && len(readyNodes) >= a.MaxNodesTotal {
 		glog.V(1).Info("Max total nodes in cluster reached")
+	} else if getOldestCreateTime(unschedulablePodsToHelp).Add(unschedulablePodTimeBuffer).After(currentTime) {
+		// The assumption here is that these pods have been created very recently and probably there
+		// is more pods to come. In theory we could check the newest pod time but then if pod were created
+		// slowly but at the pace of 1 every 2 seconds then no scale up would be triggered for long time.
+		glog.V(1).Info("Unschedulable pods are very new, waiting one iteration for more")
```

Comment:
I think (not sure though) that without a return here we can actually do a scale-down in this loop. We don't consider unschedulable pods in scale-down logic (because previously we would only get there if len(unschedulablePodsToHelp) > 0 or no existing node group can fit any of the unschedulable pods).

Reply:
Fixed.